[ci] Don't pin versions of specific packages in CI

drybjed opened the pull request:
Continuous Integration should use newest versions of packages to check if the code is compatible.

drybjed committed:
[ci] Don't pin versions of specific packages in CI

Continuous Integration should use newest versions of packages to check
if the code is compatible.

drybjed committed:
Change the way DebOps checks current user

The 'os.getlogin()' function relies on the glibc 'getlogin()' function,
which requires a control terminal (tty, pty) to work. In an environment
without the control terminal (systemd service, remote execution) it
results in the 'OSError: [Errno 6] No such device or address' exception.

Switching to 'getpass.getuser()' function is a recommended alternative.

drybjed committed:
[ci] Set no_log=false for all tasks in test env

This reveals all passwords and sensitive data inside of the DebOps CI
Pipeline on GitHub Actions, but since all passwords are randomly
generated, this shouldn't be a security risk.

drybjed committed:
[ci] Use system-wide Python interpreter for tests

This hopefully avoids an issue with Ansible executed against localhost
to randomly switch to a different Python environment. We want to be able
to use Python modules installed via APT, for example for MariaDB/MySQL
access.

drybjed committed:
[ci] Increase timeout for documentation builds

BerkhanBerkdemir commented on requirements-integration.txt:
```diff
@@ -1,28 +1,29 @@
 
 # List of Python packages used by GitHub CI pipeline for continuous integration
 
-ansible==8.0.0a1
```
This was done because in the past we thought Ansible broke the API. See https://github.com/debops/debops/issues/2442#issuecomment-1787160157

drybjed committed:
[docs] Remove 'html_theme_path' variable

This variable causes a warning on newer Sphinx versions, which results
in an error in the CI environment and fails the checks.

drybjed committed:
[ci] Use Sphinx v5.3.0 in test environment

The newest Sphinx version (v8.1.3) is very slow, no idea why for now.
Let's try an older Sphinx in CI to see if this helps make the tests
faster.
